src/c++/perf_analyzer/infer_context.cc

```diff
@@ -242,9 +246,12 @@ InferContext::AsyncCallbackFuncImpl(cb::InferResult* result)
       thread_stat_->cb_status_ = result_ptr->Id(&request_id);
       const auto& it = async_req_map_.find(request_id);
       if (it != async_req_map_.end()) {
+        it->second.end_times.push_back(end_time_async);
+      }
+      if (final_response) {
         thread_stat_->request_timestamps_.emplace_back(std::make_tuple(
-            it->second.start_time_, end_time_async, it->second.sequence_end_,
-            it->second.delayed_));
+            it->second.start_time_, it->second.end_times,
```

Comment:
this no longer has the iterator validation that it used to have. `it` may be invalid at this point now.

Reply:
thank you for catching that. 
moved the final response inside the iterator validation